lib/stream.js

```diff
@@ -45,37 +45,50 @@ Stream.Stream = Stream;
 try {
   const types = require('util').types;
   if (types && typeof types.isUint8Array === 'function') {
-    Stream._isUint8Array = types.isUint8Array;
+    // Stream._isUint8Array = types.isUint8Array;
+    Stream._isArrayBufferView = types.isArrayBufferView;
   } else {
     // This throws for Node < 4.2.0 because there's no util binding and
     // returns undefined for Node < 7.4.0.
-    Stream._isUint8Array = process.binding('util').isUint8Array;
+    // Stream._isUint8Array = process.binding('util').isUint8Array;
+    Stream._isArrayBufferView = process.binding('util').isArrayBufferView;
   }
 } catch (e) {
 }
 
-if (!Stream._isUint8Array) {
-  Stream._isUint8Array = function _isUint8Array(obj) {
-    return Object.prototype.toString.call(obj) === '[object Uint8Array]';
+if (!Stream._isArrayBufferView) {
+  Stream._isArrayBufferView = function _isArrayBufferView(obj) {
+    return [
+      '[object Int8Array]',
+      '[object Uint8Array]',
+      '[object Uint8ClampedArray]',
+      '[object Int16Array]',
+      '[object Uint16Array]',
+      '[object Int32Array]',
+      '[object Uint32Array]',
+      '[object Float32Array]',
+      '[object Float64Array]',
+      '[object DataView]'
+    ].includes(Object.prototype.toString.call(obj));
   };
 }
 
 const version = process.version.substr(1).split('.');
 if (version[0] === 0 && version[1] < 12) {
-  Stream._uint8ArrayToBuffer = Buffer;
+  Stream._typedArrayToBuffer = Buffer;
 } else {
   try {
     const internalBuffer = require('internal/buffer');
-    Stream._uint8ArrayToBuffer = function _uint8ArrayToBuffer(chunk) {
+    Stream._typedArrayToBuffer = function _typedArrayToBuffer(chunk) {
       return new internalBuffer.FastBuffer(chunk.buffer,
                                            chunk.byteOffset,
                                            chunk.byteLength);
     };
   } catch (e) {
   }
 
-  if (!Stream._uint8ArrayToBuffer) {
-    Stream._uint8ArrayToBuffer = function _uint8ArrayToBuffer(chunk) {
+  if (!Stream._typedArrayToBuffer) {
+    Stream._typedArrayToBuffer = function _typedArrayToBuffer(chunk) {
       return Buffer.prototype.slice.call(chunk);
```

Comment:
This will likely not work with anything besides `Uint8Array`. So any other TypedArray would fail.

Reply:
updated `._typedArrayToBuffer` to use `Buffer.from()`.